rootfs/api/models/app.py

```diff
@@ -540,7 +550,13 @@ def deploy(self, release, force_deploy=False, rollback_on_failure=True):  # noqa
             ]
 
             try:
-                async_run(tasks)
+                if migration_pod_exists == None:
+                    async_run(tasks)
+                else:
+                    # Deploy the "migration" pod first. If it fails the readiness checks the whole deployment will fail
+                    async_run( [tasks[0]])
```

Comment:
at this point, are you sure that tasks has at least one element?

Reply:
If we got to this branch of `if/else` then we are usre that the `migration` pod exists and that we have explicitly set it as the first element of `tasks`. Also, since the method `async_run` defined [here](https://github.com/teamhephy/controller/blob/master/rootfs/api/utils.py#L153) will just return if the array is empty, we are safe in the extreme case that only the migration pod exists, as the `async_run(tasks[1:])` won't result to exceptions.

Reply:
yeah, alright